data-processing/scripts/dsk/convert_dsk_to_jsonl.py

```diff
@@ -0,0 +1,227 @@
+"""
+Convert from various file formats (pdf, docx, etc.).
+
+To this format:
+{
+    "id": "...",             # MANDATORY: source-specific identifier
+    "text": "foo",           # MANDATORY: textual content of the document
+    "source": "...",         # MANDATORY: source of the data, such as peS2o, common-crawl, etc.
+    "added": "...",          # OPTIONAL: timestamp we acquired this data (time file was created), specified as YYYY-MM-DDTHH:MM:SS.TIMEZONE e.g 2021-04-13T12:52:46.000Z
+    "created": "..."         # OPTIONAL: timestamp when orig document was created (best-guess if not available), should be specified as a range; "YYYY-MM-DDTHH:MM:SS.TIMEZONE, YYYY-MM-DDTHH:MM:SS.TIMEZONE"
+    "metadata": {            # OPTIONAL: source-specific metadata
+        "sub-source": "...", # OPTIONAL: E.g. "newspaper_ocr"
+        ...
+    }
+}
+"""
+
+import gzip
+import json
+import re
+from dataclasses import asdict
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+from docling.datamodel.document import TableItem, TextItem
+from joblib import Parallel, delayed
+from loguru import logger
+from pdfminer.high_level import extract_text as extract_pdf_text
+from pypandoc import convert_file
+from tqdm import tqdm
+from trafilatura import extract as extract_html_text
+from typer import Typer
+from utils import (
+    build_document_converter,
+    build_metadata,
+    create_JSONL,
+    find_near_duplicates,
+    make_unique,
+    remove_newlines,
+)
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+APP = Typer(name="2JSONL Converter")
+
+
+# UNUSED - USING DOCLING INSTEAD
+def process_pdf(file_path: Path, source: str, **kwargs: dict[str, Any]) -> str:
+    text = extract_pdf_text(file_path)
+    text = re.sub(r"(\n\s)+", "\n", text)
+    metadata = {**kwargs, **build_metadata(file_path)}
+    return json.dumps(asdict(create_JSONL(text, source, metadata)), ensure_ascii=False)
+
+
+# UNUSED - USING DOCLING INSTEAD
+def process_docx(file_path: Path, source: str, **kwargs: dict[str, Any]) -> str:
+    text = convert_file(file_path, to="plain", format="docx")
+    text = re.sub(r"(\n\s)+", "\n", text)
+    metadata = {**kwargs, **build_metadata(file_path)}
+    return json.dumps(asdict(create_JSONL(text, source, metadata)), ensure_ascii=False)
+
+
+def process_html(file_path: Path, source: str, **kwargs: dict[str, Any]) -> str:  # noqa: ARG001
+    """Read a single HTML file and build a JSONL object. Uses Trafilatura for the extraction.
+
+    Args:
+        file_path (Path): Path to the HTML file
+        source (str): What is the data source (most likely DSK client)
+        **kwargs (dict): Additional arguments
```

Comment:
I believe this is a general comment, but let us not duplicate type hints

```suggestion
        file_path: Path to the HTML file
        source: What is the data source (most likely DSK client)
        **kwargs: Additional arguments
```

can lead to errors where the type hints disagree.